lib/screens/explore_screens/plant_seeds/plant_seeds_screen.dart

```diff
@@ -30,24 +31,15 @@ class PlantSeedsScreen extends StatelessWidget {
           listenWhen: (_, current) => current.pageCommand != null,
           listener: (context, state) {
             final pageCommand = state.pageCommand;
-
             if (pageCommand is ShowPlantSeedsSuccess) {
-              showDialog<void>(
-                context: context,
-                barrierDismissible: false,
-                builder: (_) {
-                  return BlocProvider.value(
-                    value: BlocProvider.of<PlantSeedsBloc>(context),
-                    child: const PlantSeedsSuccessDialog(),
-                  );
-                },
-              );
+              const PlantSeedsSuccessDialog().show(context, BlocProvider.of<PlantSeedsBloc>(context));
             }
             if (pageCommand is ShowError) {
               eventBus.fire(ShowSnackBar(pageCommand.error.localizedDescription(context)));
             }
           },
-          builder: (context, PlantSeedsState state) {
+          buildWhen: (previous, current) => previous.pageState != current.pageState,
```

Comment:
In fact you don't need remove the BlocConsumer here you only need add the buildWhen clasuse. I already re-add the BlocConsumer.